gravitee-node-kubernetes/src/main/java/io/gravitee/node/kubernetes/keystoreloader/KubernetesConfigMapKeyStoreLoader.java

```diff
@@ -97,8 +92,41 @@ protected Completable init() {
         final List<Completable> locationObs = resources
             .keySet()
             .stream()
-            .map(location -> kubernetesClient.get(ResourceQuery.<ConfigMap>from(location).build()).flatMapCompletable(this::loadKeyStore))
-            .collect(Collectors.toList());
+            .map(location -> {
+                if (CERTIFICATE_FORMAT_PEM_REGISTRY.equals(location)) {
```

Comment:
I think you should create a new keystoreloader for the pem-registry. Maybe you will need a abstract class for config map. This class despite dealing with config map serve a very different purposes.